### PR TITLE
Ansible role for joining rhel instances to AD

### PR DESCRIPTION
**Pull request by pavmoj**
- Gets linux joining account details from systems manager
- Installs required packages
- Links machine to AD

### CODE REVIEW

**Review by jnq — APPROVED** [review]
lgtm 👍

**Review by robertsweetman — APPROVED** [review]

File: ansible/roles/join-ad-linux/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+service_account_ssm_passwords:
+  service_account:
+    parameter: "/join_domain_linux_service_account/passwords"
+    users:
+      - username: auto
```
Comment (drobinson-moj) [inline ansible/roles/join-ad-linux/defaults/main.yml:6]: If you set auto, ansible will auto generate a password for you.  You probably don't want this.  Just set blank, e.g.
- username:
- password:
Then manually update the SSM param via console/CLI and set to json "{"username": "myusername", "password": "mypassword"} before running the ansible

**Review by drobinson-moj — CHANGES_REQUESTED** [review]
very minor tweak

File: ansible/roles/join-ad-linux/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+service_account_ssm_passwords:
+  service_account:
+    parameter: "/join_domain_linux_service_account/passwords"
```
Comment (drobinson-moj) [inline ansible/roles/join-ad-linux/defaults/main.yml:4]: Longer term, we'll probably want to store other AD passwords in SSM or Secrets Manager.  Might be better to have a path like this
/ad_domain/azure.noms.root/join_domain_creds

**Review by drobinson-moj — APPROVED** [review]

**Pull request merged**